### PR TITLE
QueryActionComponent: Add dataSourceRef to ActionComponetProps

### PR DESCRIPTION
**Pull request by bossinc**
Continuing from #39453 
We are adding the `DatasourceRef` to the `ActionComponentProp` so we can get the data source when creating an `ActionComponent`. We are using the for Recorded Queries in Enterprise.

### CODE REVIEW

**Review by hugohaggmark — APPROVED** [review]
LGTM 🎉

**Comment by torkelo** [comment]
We are about to change DataSourceRef to be an object (with name & uid), https://github.com/grafana/grafana/pull/33817 

Can we hold this until after that is merged?

**Comment by MasslessParticle** [comment]
@torkelo This PR blocks enterprise feature work until it lands, can we merge it before #33817 is merged?

Because we're using the `DataSourceRef`, type my understanding is that merging this shouldn't be disruptive when the refactor lands. @ryantxu Is that correct? 

It'll break us on the enterprise side, but the changes to fix are minimal.

**Comment by torkelo** [comment]
Yea you are probably right

File: public/app/features/query/components/QueryActionComponent.ts
Comment (torkelo) [inline public/app/features/query/components/QueryActionComponent.ts:8]: actually what makes more sense to send here is dataSource: DataSourceInstanceSettings, 
https://github.com/grafana/grafana/blob/ef4a76c8e6bfbef403294dbb9d5bc6cff6902108/public/app/features/query/components/QueryEditorRow.tsx#L42
 
That is passed to the QueryEditorRow, then you wont have to look it up again

**Review by ryantxu — APPROVED** [review]

**Pull request merged**
